docs/docs.sh

```diff
@@ -0,0 +1,142 @@
+#!/bin/bash
+
+# Get the 1st argument passed to the script and convert it to lowercase
+arg1=$(echo $1 | tr '[:upper:]' '[:lower:]')
+arg2=$(echo $2 | tr '[:upper:]' '[:lower:]')
+
+
+# Build the MkDocs container which will be used to build/serve the MkDocs project
+build_container () {
+    echo "Building the container..."
+    docker build -t mkdocs:latest ./mkdocs/
+    if [ $? -ne 0 ]; then
+        echo "Failed to build the container"
+        exit 1
+    fi
+}
+
+drift_dev () {
+    echo "Running Drift Dev..."
+    dart run drift_dev schema steps lib/snippets/migrations/exported_eschema/ lib/snippets/migrations/schema_versions.dart
+    if [ $? -ne 0 ]; then
+        echo "Failed to run Drift Dev"
+        exit 1
+    fi
+    dart run drift_dev schema generate --data-classes --companions lib/snippets/migrations/exported_eschema/ lib/snippets/migrations/tests/generated_migrations/
+    if [ $? -ne 0 ]; then
+        echo "Failed to run Drift Dev"
+        exit 1
+    fi
+}
+
+serve_mkdocs () {
+    echo "Running MkDocs serve..."
+    docker run --rm -p 9000:9000 -v $(pwd):/docs --user $(id -u):$(id -g) mkdocs:latest serve -f /docs/mkdocs/mkdocs.yml -a 0.0.0.0:9000
+}
+
+if [ $arg1 == "build" ]; then
+
+    echo "Building the project..."
+
+    # Remove the `build` directory if it already exists
+    rm -r ./deploy
+    rm -r ./build
+
+    # Activate the webdev command
+    dart pub global activate webdev
+
+    drift_dev
+
+    # The below command will compile the dart code in `/web` to js & run build_runner
+    webdev build -o web:build/web -- --delete-conflicting-outputs
+    if [ $? -ne 0 ]; then
+        echo "Failed to build the project"
+        exit 1
+    fi
+
+    # Run the build_runner command to generate files in the `test` directory
+    dart run build_runner build --delete-conflicting-outputs
+    if [ $? -ne 0 ]; then
+        echo "Failed to build the project"
+        exit 1
+    fi
+
+    # Remove some files that are not needed
+    rm -r ./build/web/.dart_tool
+    rm -r ./build/web/packages
+    rm ./build/web/.build.manifest
+    rm ./build/web/.packages
+
+    # Move the contents of `build` to `docs` without overwriting the `docs`
+    mv -f ./build/web/* ./docs
+
+    # Build the flutter web project
+    cd ../examples/app
+    flutter pub get
+    if [ $? -ne 0 ]; then
+        echo "Failed to build the example project"
+        exit 1
+    fi
+    # Run build_runner to generate files in the `test` directory
+    dart run build_runner build --delete-conflicting-outputs
+    if [ $? -ne 0 ]; then
+        echo "Failed to build the example project"
+        exit 1
+    fi
+
+    flutter build web --base-href "/examples/app/" --no-web-resources-cdn
```

Comment:
Nice 👍 Do we also have a way to disable the service worker?

Reply:
I don't understand what you mean

Reply:
Flutter generates a service worker that caches assets for apps so that they load faster on subsequent visits. The worker doesn't properly realize when assets have changed though, or wants us to include a version number everywhere. So we should disable the service worker in the example app, no worker should run on the documentation website.

Reply:
Done